### DISCUSSION TITLE
Update trait with trait.set() dont work

### DISCUSSION THREAD

**Discussion by capoia** [GrapesJS/grapesjs#5582] [Q&A]
Hello, how are you?

I have a trait called form, which is made up of a list of forms that come from my backend, when dragging my component to the canvas a modal (from sweetalert2) opens, when the person selects the form and clicks on "ok" I I would need to update the form trait, which in turn would update the form property (I'm using changeProp:true) which in turn emits a change:form which is where all my code happens.
The problem is: I tried to update the trait in every way, it even shows that the trait is updated, but visually it does not update the select, nor does it emit the event.

I managed to trigger the change:form manually, but I feel like it's not like that, I need to update the trait and it does the rest of the cascading effect.

![CleanShot 2023-12-26 at 10 45 32](https://github.com/GrapesJS/grapesjs/assets/21159928/073a21f6-6dce-425c-8b32-03686ec1d8ed)
![CleanShot 2023-12-26 at 10 45 51](https://github.com/GrapesJS/grapesjs/assets/21159928/ac256fa1-eb98-4ae6-be00-1547b470ba5e)
![CleanShot 2023-12-26 at 10 46 33](https://github.com/GrapesJS/grapesjs/assets/21159928/bcf57e14-91af-442b-849b-da411759bc6f)

https://github.com/GrapesJS/grapesjs/assets/21159928/8f2e209d-56f9-4c7d-9cf2-bf498a34a9bb

**Comment by capoia** [comment]
anyone can help with this?

**Comment by artf** [comment]
It should work by updating the component `form` property as you created a trait for it.
```js
.then((result) => {
 // ...
 this.set({ form: form.id });
})
```
